src/gcinfo/gcinfoencoder.cpp

```diff
@@ -1217,7 +1217,7 @@ void GcInfoEncoder::Build()
         // (after, of course, adding the size of the call instruction to get the return PC).
         callSite += m_pCallSiteSizes[callSiteIndex] - 1;
 
-        _ASSERTE(DENORMALIZE_CODE_OFFSET(NORMALIZE_CODE_OFFSET(callSite)) == callSite);
+        _ASSERTE(DENORMALIZE_CODE_OFFSET(NORMALIZE_CODE_OFFSET(callSite + 1)) == (callSite + 1));
```

Comment:
Yes, as @parjong says, the encoding intentionally expects an odd number for ARM/ARM64:
https://github.com/dotnet/coreclr/blob/master/src/gcinfo/gcinfoencoder.cpp#L1210-L1218
https://github.com/dotnet/coreclr/blob/master/src/vm/gcinfodecoder.cpp#L397-L401

I opened #4497 to address the issue of saving bits in the encoding on ARM/ARM64,